test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/DefaultModelMetadataTest.cs

```diff
@@ -411,8 +411,72 @@ public void PropertiesEnumerationEvaluatedOnce()
             Assert.Equal(firstPropertiesEvaluation, secondPropertiesEvaluation);
         }
 
+        [Fact]
+        public void IsReadOnly_ReturnsFalse_ForType()
+        {
+            // Arrange
+            var detailsProvider = new EmptyCompositeMetadataDetailsProvider();
+            var provider = new DefaultModelMetadataProvider(detailsProvider);
+
+            var key = ModelMetadataIdentity.ForType(typeof(int[]));
+            var cache = new DefaultMetadataDetails(key, new object[0]);
+
+            var metadata = new DefaultModelMetadata(provider, detailsProvider, cache);
+
+            // Act
+            var isReadOnly = metadata.IsReadOnly;
+
+            // Assert
+            Assert.False(isReadOnly);
+        }
+
+        [Fact]
+        public void IsReadOnly_ReturnsTrue_ForPrivateSetProperty()
+        {
+            // Arrange
+            var detailsProvider = new EmptyCompositeMetadataDetailsProvider();
+            var provider = new DefaultModelMetadataProvider(detailsProvider);
+
+            var key = ModelMetadataIdentity.ForType(typeof(Foo));
+            var cache = new DefaultMetadataDetails(key, new object[0]);
+
+            var metadata = new DefaultModelMetadata(provider, detailsProvider, cache);
+
+            // Act
+            var isReadOnly = metadata.Properties["Bar"].IsReadOnly;
+
+            // Assert
+            Assert.True(isReadOnly);
+        }
+
+        [Fact]
+        public void IsReadOnly_ReturnsFalse_ForPublicSetProperty()
+        {
+            // Arrange
+            var detailsProvider = new EmptyCompositeMetadataDetailsProvider();
+            var provider = new DefaultModelMetadataProvider(detailsProvider);
+
+            var key = ModelMetadataIdentity.ForType(typeof(Foo));
+            var cache = new DefaultMetadataDetails(key, new object[0]);
+
+            var metadata = new DefaultModelMetadata(provider, detailsProvider, cache);
+
+            // Act
+            var isReadOnly = metadata.Properties["Baz"].IsReadOnly;
+
+            // Assert
+            Assert.False(isReadOnly);
+        }
+
         private void ActionMethod(string input)
         {
         }
+
+        private class Foo
+        {
+            public string Bar { private set; get; }
```

Comment:
You mean the condition is for them *to be public* right? We presumably want to ignore protected/private/internal?